test/gurka/test_landmarks.cc

```diff
@@ -108,21 +108,26 @@ TEST(LandmarkTest, TestParseLandmarks) {
   std::vector<Landmark> landmarks{};
   LandmarkDatabase db(db_path, true);
 
-  EXPECT_NO_THROW({ landmarks = db.get_landmarks_in_bounding_box(-5, 0, 0, 10); });
+  EXPECT_NO_THROW({ landmarks = db.get_landmarks_by_bbox(-5, 0, 0, 10); });
   EXPECT_EQ(landmarks.size(), 3); // A, B, D
 
   LOG_INFO("Get " + std::to_string(landmarks.size()) + " rows");
   for (const auto& landmark : landmarks) {
-    LOG_INFO("name: " + std::get<0>(landmark) +
-             ", type: " + std::to_string(static_cast<uint8_t>(std::get<1>(landmark))) +
-             ", longitude: " + std::to_string(std::get<2>(landmark)) +
-             ", latitude: " + std::to_string(std::get<3>(landmark)));
+    LOG_INFO("id: " + std::to_string(std::get<0>(landmark)) + ", name: " + std::get<1>(landmark) +
+             ", type: " + std::to_string(static_cast<uint8_t>(std::get<2>(landmark))) +
+             ", longitude: " + std::to_string(std::get<3>(landmark)) +
+             ", latitude: " + std::to_string(std::get<4>(landmark)));
   }
 
-  EXPECT_TRUE(std::get<1>(landmarks[0]) == LandmarkType::bar); // A
-  EXPECT_TRUE(std::get<0>(landmarks[0]) == "A");
-  EXPECT_TRUE(std::get<1>(landmarks[1]) == LandmarkType::restaurant); // B
-  EXPECT_TRUE(std::get<0>(landmarks[1]) == "hai di lao");
-  EXPECT_TRUE(std::get<1>(landmarks[2]) == LandmarkType::cinema); // D
-  EXPECT_TRUE(std::get<0>(landmarks[2]) == "wan da");
+  EXPECT_TRUE(std::get<2>(landmarks[0]) == LandmarkType::bar); // A
+  EXPECT_TRUE(std::get<1>(landmarks[0]) == "A");
+  EXPECT_TRUE(std::get<2>(landmarks[1]) == LandmarkType::restaurant); // B
+  EXPECT_TRUE(std::get<1>(landmarks[1]) == "hai di lao");
+  EXPECT_TRUE(std::get<2>(landmarks[2]) == LandmarkType::cinema); // D
+  EXPECT_TRUE(std::get<1>(landmarks[2]) == "wan da");
+
+  // check getting multiple landmarks by ids
+  landmarks.clear();
+  EXPECT_NO_THROW({ landmarks = db.get_landmarks_by_ids({1, 2, 3, 4}); });
+  EXPECT_EQ(landmarks.size(), 3);
```

Comment:
we should talk about allowing to pass IDs which aren't actually in the DB. I guess we could do it like that and the caller should make sure the returned size is as expected, i.e. in this case `4` doesn't exist.

Reply:
I've done a few tests to make sure the return size is as expected. since we are using `"SELECT id, name, type, X(geom), Y(geom) FROM landmarks WHERE id IN (?, ?, ?, ?)";`, adding non-existing ids to the parameter wouldn't impact on the existing ids and intended results. it won't throw an error either but I guess we don't need that :D

Reply:
> adding non-existing ids to the parameter wouldn't impact on the existing ids and intended results

Sure, there's no negative impact per se, but the caller needs to be aware that the return could be less than what it asked for (and that it does not throw an exception if it encounters a missing ID). 

That brings up another important point: before merging this, please make sure there's proper docstrings on the header function declarations. A note in the function's docstring about what I just mentioned would be good.

Reply:
done. please take a look